### ISSUE TITLE
5/1 4 Corner Presentations

### ISSUE THREAD

**Issue by lunaiwa** [Codemaxxers/Issues#76]
# 4 Corner Presentations

### Project

Our project is ....
- purpose
- functions
- ect.
add screen shots here

### Demo Code 

- screen shots and explinations

### How Our Group Works

In total we have 11 group members... We have to make sure each of us are doing our work by setting up **triangles**... Each triangle has 3-4 people within them and we use them as our mini groups within our group ect...

### Triangles
- Triangle 1 (Luna, Rachit, Tanisha)
- Triangle 2 (Theo, Finn, Emma, Grace)
- Triangle 3 (Vivian, Aliya, Justin, Shivansh)

#### Triangle 2 and 3
Cyber Game Integration (Vivian, Grace, Aliya, Emma)
- Integrating games that teach cybersecurity into the overall game system
- Created an engaging user interface where the player enters a room and can play the games directly on a cartoon computer screen --> feels more interactive

Cyber Room that is entered from main page:
![image](https://github.com/Codemaxxers/Issues/assets/82780605/ebd39867-4c61-4153-801e-cfabc6fac07e)

Computer screen, can click icons to play the games:
![image](https://github.com/Codemaxxers/Issues/assets/82780605/eff7eea3-6a84-4c57-a585-bff3f9a4dbd0)

Games displayed and played on pixel computer:
![image](https://github.com/Codemaxxers/Issues/assets/82780605/4027ebfa-b7ef-49c3-974b-d3ebf786710f)
![image](https://github.com/Codemaxxers/Issues/assets/82780605/f3875017-9e84-496e-8083-2821537eb025)

Chat Bot Feature (Shivansh and Justin)
- chat bot feature to ask questions about the game or the content of the game. Serves an assistance purpose
![image](https://github.com/Codemaxxers/Issues/assets/82780605/555d0462-3412-4b48-94f8-5f894e6aa8e6)

Working on:
- Connecting the cyber games to the map made with tiles for the other games. Instead of just clicking a link, users would be able to move their character into the room with the wasd keys. 
- Integrating a point system for the cybersecurity into the leaderboard system for other games

Current Leaderboard with just CSA points:
![image](https://github.com/Codemaxxers/Issues/assets/82780605/cfffad23-3eba-4b7b-8afa-d1b5ff2ff716)

<h1> Feature Demos 🤩</h1>

<h2>Finn</h2>

- [x] [Revamped Questions database](https://github.com/Codemaxxers/codemaxxerBackend/pull/62)
- [x] New Layout/Backend
- [x] Collaboration with Theo's inventory concept
- [x] New Menu

![image](https://github.com/Codemaxxers/Issues/assets/111464990/d1131397-1e42-496b-9d60-716af11c7418)

![image](https://github.com/Codemaxxers/Issues/assets/111464990/a3cb525f-7330-4445-8094-7a75644f9fe6)

<h2>Theo</h2>

- [x] Menu bar revamp to show player data
- [x] Quest bar functional
- [ ] Backend user adding for quests
- [x] Small menu trasfer screen for game
- [x] Single/Multiplayer menu
- [x] Backend method for inventory

<img width="670" alt="Screenshot 2024-04-18 at 9 02 52 AM" src="https://github.com/Codemaxxers/Issues/assets/47646488/5a857309-b7a4-40a1-b2b0-4edc5b9f5434">

<img width="1512" alt="Screenshot 2024-04-18 at 8 53 45 AM" src="https://github.com/Codemaxxers/Issues/assets/47646488/7e8ecda1-c745-460c-949f-e35c9bfa2aa9">

<img width="1512" alt="Screenshot 2024-04-18 at 8 54 23 AM" src="https://github.com/Codemaxxers/Issues/assets/47646488/0ce7a73d-3be1-47a3-b179-6afd49a396ab">

<img width="1512" alt="Screenshot 2024-04-18 at 8 55 24 AM" src="https://github.com/Codemaxxers/Issues/assets/47646488/f308a7fb-f52c-4861-a0ae-9a1c70f98211">

<img width="1512" alt="Screenshot 2024-04-18 at 8 56 13 AM" src="https://github.com/Codemaxxers/Issues/assets/47646488/59754ca0-bb2f-414d-9e91-0a97a3a6d4d8">

<h2>Emma</h2>

- [x] make icons fixed no matter computer dimensions (computer screen button aligns w/button) 

<img width="452" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/ea793871-db1e-47e4-9738-f749aca0cf3c">

- [x] upgrade CSS styling for phishing attack (background turns green when user answers correctly for instant feedback) and password game 

<img width="500" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/da3eb1a3-4676-4180-9fe1-f3cb4f18a072">

<img width="500" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/8fdb2253-2363-4142-b7a7-476824653c38">

<img width="500" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/46be329a-5b91-4f16-9c37-4977e8dec8a6">

<img width="500" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/ecc9bcfc-a5df-49b7-9641-4cabea526c3c">

- [ ] make collision so user can enter cyber house 
- [ ] connect cyber game points with game leaderboard

<h2>Grace</h2>

- [x] backend database of terms (csp, csa, and cyber)
- [x] GET function to obtain random term on front end
- [ ] gravity CSS, make it bootiful
- [ ] figure out how to make it so that new term is taken every time a new "rock" falls

<img width="1440" alt="Screenshot 2024-04-17 at 7 54 15 PM" src="https://github.com/Codemaxxers/Issues/assets/94924172/568b8fe7-ce8e-4229-9fc8-2a263a696012">

<img width="1440" alt="Screenshot 2024-04-17 at 8 29 41 PM" src="https://github.com/Codemaxxers/Issues/assets/94924172/3cebd459-5fd2-44ca-8aff-78f6a24f084b">

<h1> Pull Requests 🤝</h1>

<img width="1278" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/fd2993f8-5ae8-4ebf-8f4d-db8961609d75">

<img width="1274" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/9ff8ed20-a4f7-4d2c-8934-abc7cc513297">

<h1>Analytics</h1>

<img width="944" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/1abf8715-fc88-4a96-9d36-00c433cc33de">

<h1> Active Team Feedback & Encouragement 😄👍</h1>

Screenshots from our "Triangles" group chat throughout March & April 

<img width="500" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/80c6c70d-6417-48b7-b32a-f0cd6406785c">

<img width="500" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/8d676bfb-fd56-43dc-be6a-bc46812eea7b">

<img width="386" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/d77fc7ba-e26c-4b21-9ca7-0af927a32c0b">

<img width="403" alt="image" src="https://github.com/Codemaxxers/Issues/assets/104966589/00048737-2ecf-47e3-a533-3569b9a18df6">

**Comment by akshat122805** [comment]
Rachita Love Story

Rift - a coding game that helps users learn code and level up to learn other computing fundamentals

**Glows**

1. Great dashboardm quest log, and level system. The game looks great frontend wise, and it seems like a professional game. The layout and functionality of the game is great.

2. The overall concept of the game is very well constructed and it seems that users can actually learn code. Also a variety of games, having three games is really cool.

**Grows**

1. Connection issues, Ditto seemed to be tough to work for this group
2. Presentation could be more cohesiveness 

The project is really good

**Comment by VishnuAravind12** [comment]
Rift
- A coding game that teaches people on the principles of computing.

Glows
- Beautiful interface and sprite work, the UI is very intuitive ad the battle mechanics make perfect sense.
- The features like the quest log, the the player stats page, and one can gain armor and weapons.
- This solves a really prevalent problem of not knowing computing fundamentals in an intuitive way.

Grows
- A think the presentation was not very "together".
- Speakers should project their voice more, they were not looking at the audience while talking.

**Comment by rohinsood** [comment]
# Overview
- Coding game, level up through topics
 - Quest log, level system, stats
 - Level up by fighting NPCs
- Spits fireball
# Glows
- User interaction with the project was very good
- Use specific game tasks with ease
# Grows
- Not enough diversity in ideas
- Presentation could be more put together

**Comment by Tirth-Thakkar** [comment]
Very good system with a comprehensive level progress system and combat to understand and demonstrate educational topics in an engaging way that would cover the main points of the lesson. Good map dynamic and art style that allows for a unified style to demo various different features and concepts from within the world and allow a user to continue to interact with the world as a whole. There could be the additional element of adding a more comprehensive art style to the mini-games and adding a learning element as they currently feel a tad out of place such as coding challenges or other similar aspects. There could be more focus on the multiplayer aspect as the idea currently feels a tad under developed with no clear presentation available.

**Comment by Henerystone** [comment]
### Glows and Grows:
- The project is high-leveled and everything about it, including the art style to the game concepts are much like a real RPG game.
- The team as a whole is also very interconnected and this aspect shows in the smoothness of the presentation.
- There is still room for improvement as there are aspects of art that can be improved. For example, the "fight scenes" have a very CGI feel to them, as if the characters and the background don't match up.
- the way the game's start screen is modeled is also very out of place as the background screens and the themes don't match up.

**Comment by h4seeb-cmd** [comment]
# Glows:
- Very cool idea
- Nice graphics on the combat aspect and game map
- Nice educational aspects, with good systems designed for player engagement
- Organized presentation

# Grows:
- Some of the minigames are a little rough graphically, email phishing game in particular. Lack of focus as well,
- Lack of multiplayer functionality as described.

**Comment by YLu-1258** [comment]
# Glows
---
- Working login system
- Very nice sprites and UI for the actual game, a lot of effort has been put into the game's design
- A lot of communication between backend and frontend
 - saving location
- Good premise: Answer CS questions in a Pokemon style battle
- Variety of minigames available in addition to the main feature, allows for a lot of people to contribute their unique features to the project
- Nice little additional site to display player data and stats.
- Very nice implementation of how the game works with collisions and the background moving
# Grows
---
- May replace the alert box with a much cleaner looking design
- Some of the textboxes in the game do not align well with the rest of the scene.
- Some of the features connected on the frontend is a lil buggy, 404 error.

**Comment by VINERAJ** [comment]
## Notes
* Can predict AP score
* Motivate students
* Game looks like pokemon
* Not working right now-stuck in collision
* quest log
* Adds to your points if you win, helps in AP scores
* Cyber Wizard-many different minigames, earns points on leaderboard
* Chatbot to answer any questions about CS the user has
* To have multiplayer functionality, needs web sockets
* Has a scrumboard and issues

## Glows
* Game looks really cool
* I like the quest and chatbot features
* Having a connection to AP is a good idea

## Grows
* Bugs need to be worked out
* Could have spoken out a bit more

**Comment by Emaad-Mir** [comment]
Glows
- Presentation is very interactive with the audience, gets us involved (in answering cs questions)
- All members in the group participated and talked about something they did
- Shows all four corners and clear connection across the frontend and backend

Grows
-  404 error on one of the features, make sure to get that fixed
- Overall presentation could have been better organized
- Could talk more about the multiplayer mode of the game

**Comment by DavidVasilev1** [comment]
## Glows

- I think that the game integrating CS knowledge is a good idea, the game is interactive and works well
- All members of the group participated and things were explained clearly
- The way the background is mapped out is cool and how all of the visual elements work together is cool as well

## Grows

- The cyber security game seems to be a bit disconnected from the rest of the game
- There were some errors with the 404 when connecting the the cyber security game and it felt to be disconnected from the rest of the actual game
- The presentation could be more integrated as well

**Comment by jm1021** [comment]
Rachita Love Story had a progressive game , tough questions, and evidences of code.   They seem to be very ad hoc on the presentations.  The 3rd period did a bit better with sharing code as  they were demoing, they did a good job of adapting to Ditto difficulties, 1st period got more hung up on challenges.

It was evident that there was a lot of work that has gone into game and passion overcame wisdom in presentation of it.   Finn and Theo just dove into game and where doing traditional demo "only" presentation. In 3rd period session, Rachit was speaking and Tanisha was showing corresponding code, this was a start to something big!  Think about entry point to discussion,  discuss a "short" plan, show a "short" demo, and "show"  behind the scenes logs, database change, or code written illustrate work.  Work the 2, 3, 4 corners, then move on. 

Glows:  After seeing demo and looking at above there is some solid Game and frontend to Backend interactions.  Player obtaining assets and managing in Java looks fantastic.

Grows: The House feature was totally overlooked and people involved did not look like they did any work.  There was no passion in this area.  The voices start to talk at the end of presentation, but other than showing UI integration it was hard to follow.  This is were several teams could be ready to present and rotate through features.
